tools/bin/mbedtls-run-tests

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+"""Run Mbed TLS unit tests.
+
+By default, run all test casess. If at least one of --prefix, --regex or
+--substring is specified, only run the test cases that match any of these
+options.
+"""
+
+import argparse
+import os
+import re
+import subprocess
+import tempfile
+
+
+class TestCaseFilter:
+    """Test case filter."""
+
+    def __init__(self, options):
+        """Set up a test case filter.
+
+        See `main` for what options are valid.
+        """
+        include = ''
+        for string in options.substring:
+            include += '|.*' + re.escape(string)
```

Comment:
probably safer to put round brackets around string - I think as-is, this would break if string contains "|"?

Reply:
Why would parentheses be useful? `re.escape` escapes special characters — and if it didn't parentheses wouldn't help.

Reply:
Good point, I missed that re.escape would deal with it